python/lsst/daf/butler/core/dimensions/_records.py

```diff
@@ -170,23 +171,22 @@ def direct(
 
         This method should only be called when the inputs are trusted.
         """
-        _recItems = record.items()
+        # This method requires tuples as values of the mapping, but JSON
+        # readers will read things in as lists. Be kind and transparently
+        # transform to tuples
+        _recItems = {k: v if type(v) != list else tuple(v) for k, v in record.items()}  # type: ignore
```

Comment:
```suggestion
        _recItems = {k: tuple(v) for k, v in record.items()}  # type: ignore
```
?

I checked that
```
a = (1, 2)
assert tuple(a) is a
```
passes, so any conditional we write in Python is unlikely to beat the compiled `tuple` constructor itself on speed.

Reply:
but not all of the things in record are lists, and we cant convert strings, numbers, et al into tuples

Reply:
Ah, I misunderstood what these elements were; definitely ignore my previous comment.  I didn't actually think _any_ of the fields of a `DimensionRecord` could be `list` _or_ `tuple`, though; just `int`, `str`, `float`, `Timespan`, or `ConvexPolygon`.

Reply:
Yeah https://github.com/lsst/daf_butler/blob/146dbb4537c4e3d171fc5a1c4bdb629136a93735/python/lsst/daf/butler/core/dimensions/_records.py#L131, tuple comes back as int, this is right from the raw python parsing

Reply:
Ah, ok, that must be the `Timespan`, then.  Thanks for explaining.